### PR TITLE
Do not report populate switch diagnostic when switch type is unknown

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Noticed while investigating https://github.com/dotnet/roslyn/issues/61281

Crash repro:
```cs
class C
{
    void M()
    {
        switch[||]
    }
}
```

It doesn't seem that this crash can happen for switch expressions (at least I couldn't find one), so test is only for switch statements.

### CODE REVIEW

**Comment by DoctorKrolic** [comment]
Can someone review this PR please?

File: src/Analyzers/CSharp/Tests/PopulateSwitch/PopulateSwitchStatementTests.cs [hunk 1]
```diff
@@ -1221,6 +1221,19 @@ public static void Test(MyEnum? myEnumValue)
...
+        public async Task TestNotInSwitchWithUnknownType()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch[||]
+    }
 }");
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/PopulateSwitch/PopulateSwitchStatementTests.cs:1237]: can you add a test like: `var v = null switch[||]` thanks!

Comment (DoctorKrolic) [reply src/Analyzers/CSharp/Tests/PopulateSwitch/PopulateSwitchStatementTests.cs:1237]: This thing currently is parsed as a switch statement, so I've put it to the same test section

**Review by CyrusNajmabadi — APPROVED** [review]
Thanks!

**Pull request merged**